plugins/chat/assets/javascripts/discourse/components/chat-drawer.js

```diff
@@ -204,74 +163,22 @@ export default Component.extend({
     );
   },
 
-  @discourseComputed("chatStateManager.isDrawerExpanded")
-  expandIcon(expanded) {
-    if (expanded) {
-      return "angle-double-down";
-    } else {
-      return "angle-double-up";
-    }
-  },
-
-  @discourseComputed("chat.activeChannel.currentUserMembership.unread_count")
-  unreadCount(count) {
-    return count || 0;
-  },
-
   @action
   openURL(URL = null) {
     this.chat.activeChannel = null;
     this.chatStateManager.didOpenDrawer(URL);
-
-    const route = this._buildRouteFromURL(
-      URL || this.chatStateManager.lastKnownChatURL
-    );
-
-    switch (route.name) {
-      case "chat":
-        this.set("view", LIST_VIEW);
-        this.appEvents.trigger("chat:float-toggled", false);
-        return;
-      case "chat.draft-channel":
-        this.set("view", DRAFT_CHANNEL_VIEW);
-        this.appEvents.trigger("chat:float-toggled", false);
-        return;
-      case "chat.channel":
-        return this._openChannel(
-          route.params.channelId,
-          this._highlightCb(route.queryParams.messageId)
-        );
-      case "chat.channel.near-message":
-        return this._openChannel(
-          route.parent.params.channelId,
-          this._highlightCb(route.params.messageId)
-        );
-      case "chat.channel-legacy":
-        return this._openChannel(
-          route.params.channelId,
-          this._highlightCb(route.queryParams.messageId)
-        );
-    }
+    this.chatDrawerRouter.stateFor(this._routeFromURL(URL));
   },
 
-  _highlightCb(messageId) {
-    if (messageId) {
-      return () => {
-        this.appEvents.trigger("chat-live-pane:highlight-message", messageId);
-      };
-    }
-  },
+  _routeFromURL(URL) {
```

Comment:
nit, but I think it's odd to use `URL` (all caps) for something that's not a constant, should just be `url`

Reply:
yes, it's some habit I have, no big deals, I can change it :D